File: TapdaqCrossPromoExamples/MoreApps/ViewController.m
//
//  ViewController.m
//  MoreApps
//
//  Created by Nick Reffitt on 08/05/2017.
//  Copyright © 2017 Tapdaq. All rights reserved.
//

#import "ViewController.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [[Tapdaq sharedSession] setDelegate:self];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

#pragma mark - Target Actions

- (IBAction)showDebugger:(id)sender {
    [[Tapdaq sharedSession] presentDebugViewController];
}

- (IBAction)loadMoreApps:(id)sender
{
    /** 
     * You can either load the more apps popup like this:
     *
     * [[Tapdaq sharedSession] loadMoreApps];
     *
     * Or you can pass in a custom config to style the more apps popup, like so:
     */
    
    [[Tapdaq sharedSession] loadMoreAppsWithConfig:[self customMoreAppsConfig]];
}

- (IBAction)showMoreApps:(id)sender
{
    if ([[Tapdaq sharedSession] isMoreAppsReady]) {
        [[Tapdaq sharedSession] showMoreApps];
    }
}

#pragma mark - Private methods

- (TDMoreAppsConfig *)customMoreAppsConfig
{
    TDMoreAppsConfig *moreAppsConfig = [[TDMoreAppsConfig alloc] init];
    
    moreAppsConfig.headerText = @"More Games";
    moreAppsConfig.installedAppButtonText = @"Play";
    
    moreAppsConfig.headerTextColor = [UIColor whiteColor];
    moreAppsConfig.headerColor = [UIColor colorWithRed:3.f / 255.f green:3.f / 255.f blue:4.f / 255.f alpha:1.f];
    moreAppsConfig.headerCloseButtonColor = [UIColor colorWithRed:235.f / 255.f green:73.f / 255.f blue:77.f / 255.f alpha:1.f];
    moreAppsConfig.backgroundColor = [UIColor colorWithRed:33.f / 255.f green:33.f / 255.f blue:33.f / 255.f alpha:1.f];
    
    moreAppsConfig.appNameColor = [UIColor whiteColor];
    moreAppsConfig.appButtonColor = [UIColor colorWithRed:59.f / 255.f green:133.f / 255.f blue:37.f / 255.f alpha:1.f];
    moreAppsConfig.appButtonTextColor = [UIColor whiteColor];
    moreAppsConfig.installedAppButtonColor = [UIColor colorWithRed:40.f / 255.f green:90.f / 255.f blue:245.f / 255.f alpha:1.f];
    moreAppsConfig.installedAppButtonTextColor = [UIColor whiteColor];
    
    return moreAppsConfig;
}

- (void)logMessage:(NSString *)message
{
    NSString *text = [self.textView text];
    NSString *textAddition = [NSString stringWithFormat:@"%@\n\n%@", message, text];
    [self.textView setText:textAddition];
    
    NSLog(@"MoreApps App: %@", message);
}

#pragma mark - TapdaqDelegate

- (void)didLoadConfig
{
    [self.loadMoreAppsBtn setEnabled:YES];
    
    [self logMessage:@"didLoadConfig"];
}

- (void)didLoadMoreApps
{
    [self.showMoreAppsBtn setEnabled:YES];
    
    [self logMessage:@"didLoadMoreApps"];
}

- (void)didFailToLoadMoreApps
{
    [self logMessage:@"didFailToLoadMoreApps"];
}

- (void)willDisplayMoreApps
{
    [self logMessage:@"willDisplayMoreApps"];
}

- (void)didDisplayMoreApps
{
    [self.showMoreAppsBtn setEnabled:NO];
    
    [self logMessage:@"didDisplayMoreApps"];
}

- (void)didCloseMoreApps
{
    [self logMessage:@"didCloseMoreApps"];
}

@end

File: samples/TapdaqExamples/MoreAppsSwift/ViewController.swift
//
//  ViewController.swift
//  MoreAppsSwift
//
//  Created by Nick Reffitt on 16/05/2017.
//  Copyright © 2017 Tapdaq. All rights reserved.
//

import UIKit
import Tapdaq

class ViewController: UIViewController, TapdaqDelegate {

    @IBOutlet var loadMoreAppsBtn: UIButton!
    @IBOutlet var showMoreAppsBtn: UIButton!
    
    @IBOutlet var textView: UITextView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        Tapdaq.sharedSession().delegate = self
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func showDebugger(_ sender: Any?) {
        Tapdaq.sharedSession().presentDebugViewController()
    }
    
    @IBAction func loadMoreApps(sender: UIButton) {
        /**
         * You can either load the more apps popup like this:
         *
         * Tapdaq.sharedSession().loadMoreApps()
         *
         * Or you can pass in a custom config to style the more apps popup, like so:
         */
        
        Tapdaq.sharedSession().loadMoreApps(with: customMoreAppsConfig())
    }
    
    @IBAction func showMoreApps(sender: UIButton) {
        
        let isReady: Bool = Tapdaq.sharedSession().isMoreAppsReady()
        
        if isReady {
            Tapdaq.sharedSession().showMoreApps()
        }
        
    }
    
    private func logMessage(message: String) {
        
        let text: String = textView!.text
        let textAddition: String = message + "\n\n" + text
        textView!.text = textAddition
        
        print("MoreApps App: " + message)
    }
    
    private func customMoreAppsConfig() -> TDMoreAppsConfig {
        
        let moreAppsConfig: TDMoreAppsConfig = TDMoreAppsConfig.init();
        
        moreAppsConfig.headerText = "More Games";
        moreAppsConfig.installedAppButtonText = "Play";
  
        moreAppsConfig.headerTextColor = UIColor.white
        moreAppsConfig.headerColor = UIColor.init(red: 3.0 / 255.0, green: 3.0 / 255.0, blue: 4.0 / 255.0, alpha: 1.0)
        
        moreAppsConfig.headerCloseButtonColor = UIColor.init(red:235.0 / 255.0, green:73.0 / 255.0, blue:77.0 / 255.0, alpha:1.0)
        moreAppsConfig.backgroundColor = UIColor.init(red:33.0 / 255.0, green:33.0 / 255.0, blue:33.0 / 255.0, alpha:1.0)
        
        moreAppsConfig.appNameColor = UIColor.white
        moreAppsConfig.appButtonColor = UIColor.init(red:59.0 / 255.0, green:133.0 / 255.0, blue:37.0 / 255.0, alpha:1.0)
        moreAppsConfig.appButtonTextColor = UIColor.white
        moreAppsConfig.installedAppButtonColor = UIColor.init(red:40.0 / 255.0, green:90.0 / 255.0, blue:245.0 / 255.0, alpha:1.0);
        moreAppsConfig.installedAppButtonTextColor = UIColor.white
        
        return moreAppsConfig;
        
    }
    
    // TapdaqDelegate
    
    func didLoadConfig() {
        loadMoreAppsBtn.isEnabled = true
        
        logMessage(message: "didLoadConfig()")
    }
    
    func didFailToLoadConfig() {
        logMessage(message: "didFailToLoadConfig()")
    }
    
    func didLoadMoreApps() {
        showMoreAppsBtn.isEnabled = true
        logMessage(message: "didLoadMoreApps()")
    }
    
    func didFailToLoadMoreApps() {
        logMessage(message: "didFailToLoadMoreApps()")
    }

    func willDisplayMoreApps() {
        logMessage(message: "willDisplayMoreApps()")
    }
    
    func didDisplayMoreApps() {
        showMoreAppsBtn.isEnabled = false
        logMessage(message: "didDisplayMoreApps()")
    }
    
    func didCloseMoreApps() {
        logMessage(message: "didCloseMoreApps()")
    }
    
}

